### PR TITLE
documentation fixes

### PR DESCRIPTION
**Pull request by alvinthai**

### CODE REVIEW

File: docs/source/api_docs_for_geeks.rst [hunk 1]
```diff
@@ -10,94 +10,91 @@ auto_ml
   :param type_of_estimator: Whether you want a classifier or regressor
   :type type_of_estimator: 'regressor' or 'classifier'
   :param column_descriptions: A key/value map noting which column is ``'output'``, along with any columns that are ``'nlp'``, ``'date'``, ``'ignore'``, or ``'categorical'``. See below for more details.
-  :type column_descriptions: dictionary, where each attribute name represents a column of data in the training data, and each value describes that column as being either ['categorical', 'output', 'nlp', 'date', 'ignore']. Note that 'continuous' data does not need to be labeled as such (all columns are assumed to be continuous unless labeled otherwise).
```
Comment (ClimbsRocks) [inline docs/source/api_docs_for_geeks.rst]: Let's delete both the opening and closing parens here.

File: docs/source/analytics.rst [hunk 2]
```diff
@@ -70,7 +69,7 @@ Interpreting Predicted Probability Buckets for Classifiers
 
 Sometimes, it's useful to know how a classifier is doing very granularly, beyond just accuracy. For instance, pretend an expensive event (say, burning a whole batch of pastries) has a 5% chance of occurring.
 
-If you train a model, obtaining 95% accuracy looks pretty bad on the surface- it's no better htan average! And in fact, you'll probably find that most (or all) of the predictions are 0- predicting that the pastries will not burn.
+If you train a model, obtaining 95% accuracy looks pretty bad on the surface - it's no better htan average! And in fact, you'll probably find that most (or all) of the predictions are 0 - predicting that the pastries will not burn.
```
Comment (ClimbsRocks) [inline docs/source/analytics.rst]: i noticed one of my own typos while looking at this line "htan"

File: docs/source/api_docs_for_geeks.rst [hunk 3]
```diff
@@ -10,94 +10,91 @@ auto_ml
...
+  :param fl_data: [default- None] If feature_learning=True, then this is the dataset we will fit the deep learning model on. This dataset should be different than your df_train dataset.
 
   :param prediction_intervals: [default- False] In addition to predicting a single value, regressors can return upper and lower bounds for that prediction as well. If you pass True, we will return the 95th and 5th percentile (the range we'd expect 90% of values to fall within) when you get predicted intervals. If you pass in two float values between 0 and 1, we will return those particular predicted percentiles when you get predicted intervals. To get these additional predicted values, you must pass in True (or two of your own float values) at training time, and at prediction time, call ``ml_predictor.predict_intervals()``. ``ml_predictor.predict()`` will still return just the prediction.
 
+  :rtype: None. This is purely to fit the entire pipeline to the data. It doesn't return anything- it saves the fitted pipeline as a property of the ``Predictor`` instance.
```
Comment (ClimbsRocks) [inline docs/source/api_docs_for_geeks.rst]: note to self: i'll probably expand upon this to mention that you can save the trained pipeline too.

File: docs/source/api_docs_for_geeks.rst [hunk 4]
```diff
@@ -10,94 +10,91 @@ auto_ml
...
-  :type ml_for_analytics: Boolean
-
   :param optimize_final_model: [default- False] Whether or not to perform GridSearchCV on the final model. True increases computation time significantly, but will likely increase accuracy.
   :type optimize_final_model: Boolean
 
-  :param perform_feature_selection: [default- True for large datasets, False for small datasets] Whether or not to run feature selection before training the final model. Feature selection means picking only the most useful features, so we don't confuse the model with too much useless noise. Feature selection typically speeds up computation time by reducing the dimensionality of our dataset, and tends to combat overfitting as well.
+  :param perform_feature_selection: [default- True for large datasets (&ge; 100,000 rows), False for small datasets] Whether or not to run feature selection before training the final model. Feature selection means picking only the most useful features, so we don't confuse the model with too much useless noise. Feature selection typically speeds up computation time by reducing the dimensionality of our dataset, and tends to combat overfitting as well.
```
Comment (ClimbsRocks) [inline docs/source/api_docs_for_geeks.rst]: great callout here, but somehow the formatting got messed up

File: docs/source/api_docs_for_geeks.rst [hunk 5]
```diff
@@ -10,94 +10,91 @@ auto_ml
...
   For more details, please visit the "categorical_ensembling" page in the docs.
 
   :rtype: None. This is purely to fit the entire pipeline to the data. It doesn't return anything- it saves the fitted pipeline as a property of the ``Predictor`` instance.
 
 
-.. py:method:: ml_predictor.predict(prediction_rows)
+.. py:method:: ml_predictor.predict(prediction_data)
 
-  :param prediction_rows: A single dictionary, or a DataFrame, or list of dictionaries. For production environments, the code is optimized to run quickly on a single row passed in as a dictionary (taking around 1 millisecond for the entire pipeline). Batched predictions on thousands of rows at a time are generally more efficient if you're getting predictions for a larger dataset.
+  :param prediction_data: A single dictionary, or a DataFrame, or list of dictionaries. For production environments, the code is optimized to run quickly on a single row passed in as a dictionary (taking around 1 millisecond for the entire pipeline). Batched predictions on thousands of rows at a time are generally more efficient if you're getting predictions for a larger dataset.
```
Comment (ClimbsRocks) [inline docs/source/api_docs_for_geeks.rst]: note to self: mention that DFs are best for batched predictions

File: docs/source/categorical_ensembling.rst [hunk 6]
```diff
@@ -22,23 +22,19 @@ You'll need to pass in an identifier for which column of data is our `categorica
...
 3. You can still pass in any of the normal arguments for `.train()` that you know and love!
 
 Performance Notes:
-A. We train up only one global transformation_pipeline to minimize disk space when serializing the models
-B. If `feature_learning=True` is passed in, we will train up one global feature_learning model- we will NOT train up one feature_learning model per category. The model we train for each category can then decide whether and how to use the features from our feature_learning model. Since each feature_learning model has to be serialized to disk separately right now, this design decision was made to reduce complexity, and the risk of things going wrong when transferring trained models to a production environment.
-
-
-
-
+<ol type="A">
```
Comment (ClimbsRocks) [inline docs/source/categorical_ensembling.rst]: oh cool, i hadn't realized you could embed html in .rst like this!

File: docs/source/api_docs_for_geeks.rst (hunk 4 above)
Comment (alvinthai) [inline docs/source/api_docs_for_geeks.rst]: ```&ge;``` should be html code for "greater than equal to" symbol [http://turner.faculty.swau.edu/webstuff/htmlsymbols.html](http://turner.faculty.swau.edu/webstuff/htmlsymbols.html)

**Comment by alvinthai** [comment]
Happy to help!

**Comment by ClimbsRocks** [comment]
thanks again! sorry this took me so long to close out some of my thoughts you inspired. really appreciate the help- let me know if you run into anything else that could be better!

**Pull request merged**